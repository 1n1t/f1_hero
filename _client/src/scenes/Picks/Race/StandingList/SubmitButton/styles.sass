.SubmitButton
	text-align: center
	border-top: 2px solid white
	cursor: pointer

