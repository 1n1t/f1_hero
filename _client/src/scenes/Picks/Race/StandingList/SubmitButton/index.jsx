import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

const SubmitButton = ({ onClick }) => (
	<div className="btn SubmitButton" onClick={onClick} role="button" tabIndex="0">
		<i className="fa fa-check" aria-hidden="true" /> Submit
	</div>
);

SubmitButton.propTypes = {
	onClick: PropTypes.func.isRequired
};

export default SubmitButton;
