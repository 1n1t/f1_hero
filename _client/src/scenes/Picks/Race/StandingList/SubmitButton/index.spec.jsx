import r from 'Root/jest/utils/renderer';

import SubmitButton from './index';

const defaultProps = {
	onClick: jest.fn()
};

r.init(SubmitButton, defaultProps);

it('renders correctly', () => {
	const wrapper = r.render();
	expect(wrapper).toMatchSnapshot();
});

it('triggers callback when clicked', () => {
	const button = r.render();

	button.simulate('click');

	expect(defaultProps.onClick).toHaveBeenCalledTimes(1);
});
