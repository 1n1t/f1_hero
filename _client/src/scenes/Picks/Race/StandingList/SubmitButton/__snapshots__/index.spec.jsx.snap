// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="btn SubmitButton"
  onClick={[MockFunction]}
  role="button"
  tabIndex="0"
>
  <i
    aria-hidden="true"
    className="fa fa-check"
  />
   Submit
</div>
`;
